src/services/healthcareProfessionalService.ts

```diff
@@ -145,7 +171,21 @@ Promise<Result<gqlTypes.HealthcareProfessional[]>> {
                     professional.degrees.some(db => filters.degrees?.some(filter => filter == db)))
                 : filteredResults ?? []
 
-            filteredResults = professionalsFilteredByDegrees
+            const professionalsFilteredByName = hasNamesFilter
+                ? professionalsFilteredByDegrees.filter(professionals => {
+                    const filterName = filters.names ? filters.names[0] : false
```

Comment:
should check length here. technically it can exist but be empty and would break if you access by `[0]`

Reply:
We mist this one sorry! 🙇‍♀️